common/app/com/commercetools/sunrise/sessions/wishlist/DefaultWishlistInSession.java

```diff
@@ -33,17 +41,24 @@ public DefaultWishlistInSession(final Configuration configuration, final ObjectS
     }
 
     @Override
+    public Optional<WishlistViewModel> findWishlist() {
+        return session.findObjectByKey(wishlistSessionKey, WishlistViewModel.class);
+    }
+
+    @Override
     protected Logger getLogger() {
         return LOGGER;
     }
 
     @Override
-    protected void storeAssociatedData(@NotNull final ShoppingList value) {
-        session.overwriteValueByKey(wishlistIdSessionKey, value.getId());
+    protected void storeAssociatedData(@NotNull final ShoppingList wishlist) {
+        session.overwriteValueByKey(wishlistIdSessionKey, wishlist.getId());
+        session.overwriteObjectByKey(wishlistSessionKey, wishlistViewModelFactory.create(wishlist));
```

Comment:
And specially document it, of course. It will work with one application, but on multi-servers it will fail miserably if not used a more advanced caching system (unless configured properly).